File: frontend/src/model/membershipUsage.ts
/** Models for table: membershipUsage (ids are number / BIGINT) */

export class MembershipUsageRes {
  id: number = 0;
  orgId: number = 0;
  locationId: number = 0;
  customerId: number = 0;
  membershipId: number = 0;
  planId: number = 0;
  invoiceId: number = 0;
  serviceId: number = 0;
  serviceName: string = "";
  quantity: number = 0;
  type: string = "";
  status: string = "";
  usedOn: string = "";
  createdby: string = "";
  createdon: string = "";
  updatedby: string = "";
  updatedon: string = "";
}

export class CreateMembershipUsageReq {
  orgId: number = 0;
  locationId: number = 0;
  customerId: number = 0;
  membershipId: number = 0;
  planId: number = 0;
  invoiceId: number = 0;
  serviceId: number = 0;
  serviceName: string = "";
  quantity: number = 0;
  type: string = "";
  status: string = "";
  usedOn: string = "";
}

export class CreateMembershipUsageRes extends MembershipUsageRes {}

export class UpdateMembershipUsageReq {
  id: number = 0;
  orgId: number = 0;
  locationId: number = 0;
  customerId: number = 0;
  membershipId: number = 0;
  planId: number = 0;
  invoiceId: number = 0;
  serviceId: number = 0;
  serviceName: string = "";
  quantity: number = 0;
  type: string = "";
  status: string = "";
  usedOn: string = "";
}

export class UpdateMembershipUsageRes extends MembershipUsageRes {}

export class GetMembershipUsageReq {
  id: number = 0;
  orgId: number = 0;
}

export class GetMembershipUsageRes extends MembershipUsageRes {
  errorMessage: string = "";
}

export class ListMembershipUsageReq {
  orgId: number = 0;
  locationId: number = 0;
  search: string = "";
  page: number = 1;
  pageSize: number = 50;
}

export class ListMembershipUsageRes {
  items: MembershipUsageRes[] = [];
  total: number = 0;
  errorMessage: string = "";
}

export class DeleteMembershipUsageReq {
  id: number = 0;
  orgId: number = 0;
}

export class DeleteMembershipUsageRes {
  id: number = 0;
  success: boolean = false;
  errorMessage: string = "";
}

File: backend/Krios/krios/Models/MembershipUsage.cs
namespace Krios.Models.Krios
{
    public class MembershipUsage
    {
        public long id { get; set; }
        public long orgId { get; set; }
        public long locationId { get; set; }
        public long customerId { get; set; }
        public long membershipId { get; set; }
        public long planId { get; set; }
        public long invoiceId { get; set; }
        public long serviceId { get; set; }
        public string serviceName { get; set; } = "";
        public decimal quantity { get; set; }
        public string type { get; set; } = "";
        public string status { get; set; } = "";
        public DateTime? usedOn { get; set; }
        public string createdby { get; set; } = "";
        public DateTime? createdon { get; set; }
        public string updatedby { get; set; } = "";
        public DateTime? updatedon { get; set; }
    }

    public class MembershipUsageSelectReq
    {
        public long id { get; set; }
        public long orgId { get; set; }
        public long locationId { get; set; }
        public string search { get; set; } = "";
        public string status { get; set; } = "";
    }

    public class MembershipUsageDeleteReq
    {
        public long id { get; set; }
        public long orgId { get; set; }
    }
}